### PR TITLE
Simulator Improvement Task#E

### PR DESCRIPTION
**Pull request by syedomarazam**
As part of Improvement we have modified transform output file format and added more readability for the users. Please review transform output files for ABP simulator, sender, receiver and subnet modules. 

Also we have created a GUI for the project. Its a brand new repository. Please follow the instructions in readme file to test GUI. User have the option to view and simulate the results using a graphical user interface. For more details on how to use this feature, go to https://github.com/mohamedali5/ABP_Gui

### CODE REVIEW

File: data/top_model/abp_output_transform.txt [hunk 1]
```diff
@@ -1,125 +1,55 @@
...
+00:01:28:000              1                   1                   1                 out            subnet2
+00:01:28:000              1                   1                   1      ackReceivedOut            sender1
+00:01:30:000              4                   4                   1                 out      generator_con
+00:01:40:000              4                   4                   1       packetSentOut            sender1
+00:01:40:000             40                   4                   0             dataOut            sender1
+00:01:43:000             40                   4                   0                 out            subnet1
+00:01:53:000              0                   0                   0                 out          receiver1
+00:01:56:000              0                   0                   0                 out            subnet2
+00:01:56:000              0                   0                   0      ackReceivedOut            sender1
+00:02:06:000              5                   5                  NA       packetSentOut            sender1
```
Comment (shubhamagrawal6629) [inline data/top_model/abp_output_transform.txt]: If you compare with other packet sentout it is different

File: src/transform_output/transform_output.cpp [hunk 2]
```diff
@@ -68,149 +68,199 @@ void transform_output(const char *ip_file_name, const char *op_file_name) {
 	 * setting the column names for our output file.
 	 * setw() function is used to give indentation in the output file.
 	 */
-	outputfile <<setw(7)<<"Time"<<setw(20)<<"Value"<<setw(14) \
-			   <<"Port"<<setw(22)<<"Component"<<endl;
-	outputfile <<"-------------------------------------------"
-				 "---------------------------"<<endl;
+	//outputfile <<setw(7)<<"Time"<<setw(20)<<"Value"<<setw(14) \
```
Comment (shubhamagrawal6629) [inline src/transform_output/transform_output.cpp]: Commented code

File: src/transform_output/transform_output.cpp [hunk 3]
```diff
@@ -68,149 +68,199 @@ void transform_output(const char *ip_file_name, const char *op_file_name) {
 	 * setting the column names for our output file.
 	 * setw() function is used to give indentation in the output file.
 	 */
-	outputfile <<setw(7)<<"Time"<<setw(20)<<"Value"<<setw(14) \
-			   <<"Port"<<setw(22)<<"Component"<<endl;
-	outputfile <<"-------------------------------------------"
-				 "---------------------------"<<endl;
+	//outputfile <<setw(7)<<"Time"<<setw(20)<<"Value"<<setw(14) \
+	//		   <<"Port"<<setw(22)<<"Component"<<endl;
+	outputfile << setw(7) << "Time" << setw(20) << "Value" << setw(22) \
```
Comment (shubhamagrawal6629) [inline src/transform_output/transform_output.cpp]: no need of slash as the end for continuation

File: src/transform_output/transform_output.cpp [hunk 4]
```diff
@@ -68,149 +68,199 @@ void transform_output(const char *ip_file_name, const char *op_file_name) {
...
+						 * if a value is obtained other than empty then we assign it to port_value variable.
+						 */
+			            if (strlen(token_4)>2){
+							
+							in = token_4;
+							port_value = in.substr( 1 , ( in.find_first_of('}', 0) - 1) );
+			        			/** 
+								* Check Alternating Bit status,only if it is equal to 1
+								*/
+								if (port_value.length() ==1 ){
```
Comment (shubhamagrawal6629) [inline src/transform_output/transform_output.cpp]: I think there should be 4 spaces removed

**Review by shubhamagrawal6629 — CHANGES_REQUESTED** [review]

File: data/top_model/abp_output_transform.txt (hunk 1 above)
Comment (pcsmartapps) [reply data/top_model/abp_output_transform.txt]: bug has be fixed.

File: src/transform_output/transform_output.cpp (hunk 2 above)
Comment (pcsmartapps) [reply src/transform_output/transform_output.cpp]: modified as requested

File: src/transform_output/transform_output.cpp (hunk 3 above)
Comment (pcsmartapps) [reply src/transform_output/transform_output.cpp]: modified as requested

File: src/transform_output/transform_output.cpp (hunk 4 above)
Comment (pcsmartapps) [reply src/transform_output/transform_output.cpp]: Modified as requested

File: src/transform_output/transform_output.cpp (hunk 3 above)
Comment (shubhamagrawal6629) [reply src/transform_output/transform_output.cpp]: \ at the end is not required it will work without \

**Review by shubhamagrawal6629 — COMMENTED** [review]
Just remove the \ at the end and also make some changes to Gui file i mailed to you

**Comment by shubhamagrawal6629** [comment]
> Just remove the \ at the end and also make some changes to Gui file i mailed to you

Ok so for gui file just the comments with // in /* comment */ format as // is not doxygen

File: src/transform_output/transform_output.cpp (hunk 3 above)
Comment (pcsmartapps) [reply src/transform_output/transform_output.cpp]: thank you for the note. modified as per requested.

**Comment by shubhamagrawal6629** [comment]
just change the comments for maingui as mentioned above 

I am approving this in a minute

**Review by shubhamagrawal6629 — APPROVED** [review]
The files have been changed properly and every thing works fine

**Review by p-sky — APPROVED** [review]

**Pull request merged**